test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/ModelStateDictionaryTest.cs

```diff
@@ -222,53 +223,16 @@ public void ConstructorWithDictionaryParameter()
             Assert.Equal("bar", newDictionary["foo"].Value.ConvertTo(typeof(string)));
         }
 
-        [Fact]
-        public void GetFieldValidationState_ReturnsUnvalidatedIfDictionaryDoesNotContainKey()
-        {
-            // Arrange
-            var msd = new ModelStateDictionary();
-
-            // Act
-            var validationState = msd.GetFieldValidationState("foo");
-
-            // Assert
-            Assert.Equal(ModelValidationState.Unvalidated, validationState);
-        }
-
-        [Fact]
-        public void GetValidationState_ReturnsValidationStateForKey_IgnoresChildren()
-        {
-            // Arrange
-            var msd = new ModelStateDictionary();
-            msd.AddModelError("foo.bar", "error text");
-
-            // Act
-            var validationState = msd.GetValidationState("foo");
-
-            // Assert
-            Assert.Equal(ModelValidationState.Unvalidated, validationState);
-        }
-
-        [Fact]
-        public void GetFieldValidationState_ReturnsInvalidIfKeyChildContainsErrors()
-        {
-            // Arrange
-            var msd = new ModelStateDictionary();
-            msd.AddModelError("foo.bar", "error text");
-
-            // Act
-            var validationState = msd.GetFieldValidationState("foo");
-
-            // Assert
-            Assert.Equal(ModelValidationState.Invalid, validationState);
-        }
-
-        [Fact]
-        public void GetFieldValidationState_ReturnsInvalidIfKeyContainsErrors()
+        [Theory]
+        [InlineData("foo")]
+        [InlineData("foo.bar")]
+        [InlineData("[0].foo.bar")]
+        [InlineData("[0].foo.bar[0]")]
+        public void GetFieldValidationState_ReturnsInvalidIfKeyChildContainsErrors(string key)
```

Comment:
IMO we should not delete/merge the other tests simply because it is hard to see if those are being covered here.